### DISCUSSION TITLE
ckan test: Running CKAN on http://localhost:5000

### DISCUSSION THREAD

**Discussion by chenxj2020** [ckan/ckan#6025] [Q&A]
I'm trying to run the CKAN test and it doesn't work.
The initializing DB is success. 
When I run the ckan.ini, it just stopping at there, and shows no error information, like following:

[ckan.cli] Using configuration file /etc/ckan/default/ckan.ini
[ckan.config.environment] Loading static files from public
[ckan.config.environment] Loading templates from /root/ckan/lib/default/src/ckan/ckan/templates
[ckan.config.environment] Loading templates from /root/ckan/lib/default/src/ckan/ckan/templates
[ckan.cli.server] Running CKAN on http://localhost:5000
[ckan.cli] Using configuration file /etc/ckan/default/ckan.ini
[ckan.config.environment] Loading static files from public
[ckan.config.environment] Loading templates from /root/ckan/lib/default/src/ckan/ckan/templates
[ckan.config.environment] Loading templates from /root/ckan/lib/default/src/ckan/ckan/templates
[ckan.cli.server] Running CKAN on http://localhost:5000

Any answer is much appreciated!

**Comment by amercader** [comment]
This looks like you got it working! What happens when you open the browser and go to http://localhost:5000 ?

**Reply by chenxj2020** [reply]
Thank you very much for your reply, this website (http://localhost:5000) can not open, there is nothing.

The next step is expected to show Database tables created as I see from other's project. So is there error with my Postgresql？ I installed the Postgresql-13, ckan-2.9.2 on centos7. 

Thank you!

**Reply by chenxj2020** [reply]
Hi, amercader,
I didn't express my queation clearly, please look at this:
This is my result in doing the ckantest:

![ckanRun1](https://user-images.githubusercontent.com/77830569/115198418-9bd11e80-a124-11eb-851a-c5199dbf0082.jpg)

when I open the browser and go to http://localhost:5000, it shows:
![ckanRun2](https://user-images.githubusercontent.com/77830569/115198422-9d024b80-a124-11eb-93fd-10a449224075.jpg)

I don't know what's the problem in my ckan test.
Thank you.

**Reply by anuveyatsu** [reply]
@chenxj2020 it looks like CKAN is running. Can you try to access using http://127.0.0.1:5000 instead of localhost? Otherwise, it is difficult to say what's wrong from the provided information.

**Reply by chenxj2020** [reply]
Thank you, anuveyatsu,
Yes, now I can open both http://localhost:5000 and http://127.0.0.1:5000. 
But, I don't know why I can not open the website with my IP, ie., replace "127.0.0.1" with my IP.
Is there any parameter that need I to set?
Thank you very much.

**Reply by anuveyatsu** [reply]
@chenxj2020 great to know it worked! I'm not sure but I think this is a networking issue and not related to CKAN itself. I'm not aware of any parameter that can resolve it.

**Reply by chenxj2020** [reply]
Yes, it may be related to network settings. Thank you so much, @anuveyatsu.

**Reply by starship9** [reply]
I'm actually facing the same problem right now! I built from source, the cli tells me CKAN is running, but I'm unable to access `localhost:5000` or even `127.0.0.1:5000`

Here's what the terminal says:

`2023-11-27 16:31:52,217 INFO  [ckan.cli.server] Running CKAN on http://localhost:5000`

**Comment by hongjie123** [comment]
I have encountered the same problem before and it has now been resolved. You need to modify the configuration file of ckan, for example: ckan.ini
```
ckan.devserver.host = 0.0.0.0
ckan.devserver.port = 5000
```
Run ckan again and you will be able to access your site through IP: 5000.

**Reply by k3ybladewielder** [reply]
@hongjie123 This is the only modification you made ? I test here and occurs the same error related by @starship9 and @chenxj2020. I'm unable to access localhost:5000 or even 127.0.0.1:5000

**Reply by hongjie123** [reply]
Yes, when modifying the ckan.ini file and restarting ckan, the following message will appear: [ckan. cli. server] Running CKAN on http://0.0.0.0:5000
You can now access it using IP+port.

**Reply by k3ybladewielder** [reply]
@hongjie123 you installed ckan from package or from source?

**Reply by hongjie123** [reply]
Source install（https://docs.ckan.org/en/2.10/maintaining/installing/index.html#source-install）

**Comment by Andries-B** [comment]
Looks like I had a similar situation. After a fresh install from ubuntu packages (https://docs.ckan.org/en/2.10/maintaining/installing/install-from-package.html) the system was running well, except for always directing the browser to 127.0.0.1:5000. This can be changed in CKAN settings file `/etc/ckan/default/ckan.ini` . After installation there was this line:

`ckan.site_url = http://127.0.0.1:5000`

Changed this to: (it listens on port 80, so no port specification needed)

`ckan.site_url = http://<your-web-server>`

Restart and it works
